### PR TITLE
Configurable sign-editor

### PR DESCRIPTION
**Pull request by HoldYourWaffle**
Since 50f53ce you need to have a sign in your (off)hand to edit a sign using rightclick.
I made the required item configurable, even allowing for not requiring an item at all.

### CODE REVIEW

**Comment by frodare** [comment]
Thanks for your interest!  However, in this PR please only change or add code relevant to the new feature.

**Comment by HoldYourWaffle** [comment]
Thought you might want that so I already changed those things in seperate commits. I've made a "cleanup" commit reverting everything not-function related.

**Comment by frodare** [comment]
That's better, but the formatting is still changed.  When making a feature change the code style should be kept the same.  Most notably in this PR, indention is 2 spaces, not tabs.  Ideally the only changes the diff should show are the code improvements you added.

**Comment by HoldYourWaffle** [comment]
Whoops didn't even notice that. Must've been my auto-formatter.
I've fixed it now 😄

File: src/main/java/net/torocraft/signedit/SignEdit.java [hunk 1]
```diff
@@ -60,12 +68,9 @@ public void editSign(RightClickBlock event) {
...
 
-  private static boolean isHoldingSign(EntityPlayer player) {
-    for (ItemStack stack : player.getHeldEquipment()) {
-      if (stack.getItem() == Items.SIGN) {
-        return true;
-      }
-    }
+  private static boolean isHoldingEditor(EntityPlayer player) {
+    if (editor == null) return true;
+    for (ItemStack stack : player.getHeldEquipment()) if (stack.getItem() == editor) return true;
```
Comment (frodare) [inline src/main/java/net/torocraft/signedit/SignEdit.java]: use the multi-line style that was use originally, again no need to change code style for this

File: src/main/java/net/torocraft/signedit/SignEdit.java [hunk 2]
```diff
@@ -26,32 +26,40 @@
...
   @EventHandler
   public void preInit(FMLPreInitializationEvent e) {
     MinecraftForge.EVENT_BUS.register(this);
+    
+    cfg = new Configuration(new File("config/signedit.cfg"));
+    cfg.load();
+    
+    String cfgEditor = cfg.get("SignEdit", "editor", "minecraft:sign", "The player must hold this item to edit signs. Enter in the format modid:itemname. Default: 'minecraft:sign'. Use '*' to always allow editing regardless of held items.").getString();
+    
+    if (cfgEditor.equals("*")) editor = null;
```
Comment (frodare) [inline src/main/java/net/torocraft/signedit/SignEdit.java]: use multi-line if blocks

File: src/main/java/net/torocraft/signedit/SignEdit.java [hunk 3]
```diff
@@ -26,32 +26,40 @@
...
   public void editSign(RightClickBlock event) {
-    if (event.getEntityPlayer().isSneaking()) {
-      return;
-    }
-
-    if (!isHoldingSign(event.getEntityPlayer())) {
-      return;
-    }
-
+    if (event.getEntityPlayer().isSneaking() || !isHoldingEditor(event.getEntityPlayer())) return;
```
Comment (frodare) [inline src/main/java/net/torocraft/signedit/SignEdit.java]: I don't think any thing changed here to add this feature, so this should be restored

File: src/main/java/net/torocraft/signedit/SignEdit.java [hunk 4]
```diff
@@ -26,32 +26,40 @@
...
+    
     BlockPos pos = new BlockPos(event.getHitVec());
     IBlockState state = event.getWorld().getBlockState(pos);
-
-    if (state.getBlock() != Blocks.WALL_SIGN && state.getBlock() != Blocks.STANDING_SIGN) {
-      return;
-    }
-
+    
+    if (state.getBlock() != Blocks.WALL_SIGN && state.getBlock() != Blocks.STANDING_SIGN) return;
```
Comment (frodare) [inline src/main/java/net/torocraft/signedit/SignEdit.java]: same here

File: src/main/java/net/torocraft/signedit/SignEdit.java [hunk 5]
```diff
@@ -26,32 +26,40 @@
   public static final String MODID = "signedit";
   public static final String VERSION = "1.12.2-4";
   public static final String MODNAME = "SignEdit";
+  
+  public static Configuration cfg;
+  public static Item editor;
```
Comment (frodare) [inline src/main/java/net/torocraft/signedit/SignEdit.java]: these members should be private

**Review by frodare — CHANGES_REQUESTED** [review]
Just a few more minor things

**Comment by HoldYourWaffle** [comment]
I fixed the minor thingies, although I must say I think some things are a bit "unnecessary". (Double if block with same code in it, multi-line if block with only a return statement after it.)

You were completely right about the private fields though that was just an oversight from me 😄

File: src/main/java/net/torocraft/signedit/SignEdit.java [hunk 6]
```diff
@@ -38,7 +55,7 @@ public void editSign(RightClickBlock event) {
       return;
     }
 
-    if (!isHoldingSign(event.getEntityPlayer())) {
+    if (isHoldingEditor(event.getEntityPlayer())) {
```
Comment (frodare) [inline src/main/java/net/torocraft/signedit/SignEdit.java]: was the `!` removed by accident?

**Comment by frodare** [comment]
Yeah, they were "unnecessary", I just prefer to keep format / cleanup changes separate from feature and logic changes. I am not saying one way is better than than the other. :)

File: src/main/java/net/torocraft/signedit/SignEdit.java (hunk 6 above)
Comment (HoldYourWaffle) [reply src/main/java/net/torocraft/signedit/SignEdit.java]: I have literally no idea how and when that happened....
Must've been somewhere in the style-fixing but that's still weird.

**Comment by HoldYourWaffle** [comment]
Well I guess that makes sense 🤔

**Comment by frodare** [comment]
Cool, thanks! I will try to get around to building and uploading this soon, I haven't been working on these mods lately.

**Pull request merged**